webapp/app/utils/makeActionCreator.js

```diff
@@ -0,0 +1,11 @@
+export function makeActionCreator(type, ...argNames) {
+  return (...args) => {
+    let action = { type };
+    
+    argNames.forEach((arg, index) => {
+      action[argNames[index]] = args[index];
+    });
+
+    return action;
+  };
+}
```

Comment:
This is what `createActions` Does

Reply:
We discontinued the use of that middleware a while back, this is the sample implementation of redux-saga from their provided guides, please check their guides and docs, for insight into why this is implemented in this way, thanks

Reply:
Please provide clear actionable comments with examples of propossed changes, thanks